### PR TITLE
Bug/inbox todo

### PR DESCRIPTION
**Pull request by nrbunn**
#6994
### Changes

missing `ng-controller='ChatCtrl'` so it was unable to find copyToDo. Next copyToDo uses group.name in scope so created inboxCtrl to supply it.

---

UUID: 25479d8c-45e1-482c-877e-6af2ccaba643

### CODE REVIEW

**Comment by nrbunn** [comment]
hmm noticed my debounce hash is in there. let me know if that causes any issues.

**Comment by crookedneighbor** [comment]
I'm hesitant to just slap the chat controller onto the inbox, as it's really just this function that is missing: https://github.com/HabitRPG/habitrpg/blob/develop/website/public/js/controllers/chatCtrl.js#L93-L109

Doesn't seem right to expose all the other functionality of the chat controller onto the inbox.

In fact, the notes field should probably be updated to be something more specific to the inbox. Why not just add a similar function to the new inbox controller? Would also eliminate the need to create a hacky group object on `$scope`?

**Comment by nrbunn** [comment]
You know didn't even think of that. Got blinding by the issue of not having group in scope when I added the chat controller. Sorry still a noob. I will revise and send a update.

**Comment by nrbunn** [comment]
Created new copytodo function in new inbox controller based on @crookedneighbor feedback.

Might need some better word smithing for default info?
inboxtodo

**Comment by crookedneighbor** [comment]
The bracketed text before was creating a link to the party/user. This bracketted text doesn't really do anything. 

Also, I don't think the "Task from <user>" text conveys useful information about the task. I'd like to hear from @lemoness and @Alys about what they want out of copying messages to tasks.

**Comment by Alys** [comment]
If I were using this feature, I'd probably rewrite the task title myself each time, or move the entire text from the note into the title. I haven't seen any comments to indicate how others would use it.

Maybe "Message from username" as the default title, with no brackets. I don't think we can predict what the majority's preferred text would be so anything simple would do. I'd prefer "Message..." rather than "Task..." because it's a more accurate description of where the material came from, and sometimes you might be copying the message to a To-Do just to think about it, rather than to perform a task.

But feel free to ignore this if there's conflicting opinions. I don't feel strongly about it.

**Comment by lemoness** [comment]
I also like "Message From Username." @veeeeeee may have options as well, though.

**Comment by nrbunn** [comment]
inboxtodo

File: common/locales/en/groups.json [hunk 1]
```diff
@@ -110,6 +110,8 @@
     "copyMessageAsToDo": "Copy message as To-Do",
     "messageAddedAsToDo": "Message copied as To-Do.",
     "messageWroteIn": "<%= user %> wrote in <%= group %>",
+    "taskFromInbox": "<%= from %> wrote, '<%= message %>'",
```
Comment (crookedneighbor) [inline common/locales/en/groups.json]: No need for a comma here.

File: website/public/js/controllers/inboxCtrl.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+habitrpg.controller("InboxCtrl", ['$scope', '$rootScope',
+    function($scope, $rootScope) {
+      $scope.copyToDo = function(message) {
+        var taskNotes = env.t("taskFromInbox",  {
+          from: message.uuid == 'system'
```
Comment (crookedneighbor) [inline website/public/js/controllers/inboxCtrl.js]: It's confusing to have a ternary here. I think this logic should be outside of the object.

Comment (crookedneighbor) [inline website/public/js/controllers/inboxCtrl.js]: Especially given the fact that we do the exact same call below.

**Comment by crookedneighbor** [comment]
This should also have a karma test before it can be merged in.

**Comment by crookedneighbor** [comment]
Overall, looking good!

**Comment by nrbunn** [comment]
Updated. Had a fun time with tests so please review to make sure I did it correctly.

**Comment by veeeeeee** [comment]
Eek, just saw this in my email. "Message from [X]" is fine by me. Other possible variations: "Request from X" "Action item via X" or even "ToDo via X"

**Comment by crookedneighbor** [comment]
Sorry it is taking me so long to review this. I'm slowly going through the PRs now.

**Comment by crookedneighbor** [comment]
Great work! I've notated this toward your next tier.

**Pull request closed without merging**